tests/unit/containers/my-cooperations/cooperation-notes/CooperationNotes.spec.jsx

```diff
@@ -1,91 +1,93 @@
 import {
   renderWithProviders,
-  mockAxiosClient,
-  TestSnackbar
+  TestSnackbar,
+  mockAxiosClient
 } from '~tests/test-utils'
-import { vi } from 'vitest'
 import { ConfirmationDialogProvider } from '~/context/confirm-context'
 import { URLs } from '~/constants/request'
+import {
+  mockNotesData,
+  mockUpdatedNotesData,
+  mockUpdatedWithDuplicatedNoteData,
+  completeNewNote,
+  finishedMockedNotesData
+} from './CooperationNotes.constants'
+import * as useQuery from '~/hooks/use-query'
 import { getFullUrl } from '~/utils/get-full-url'
 
-const mockNotesData = [
-  {
-    _id: '65b03361bf20871d3adaeb9c',
-    text: 'Test note1 content',
-    author: {
-      _id: '6565fd508a848ff2202df79c',
-      firstName: 'User',
-      lastName: 'Test',
-      role: 'tutor'
-    },
-    isPrivate: false,
-    cooperation: '65afbd053d67b51996a67c4c',
-    createdAt: '2024-01-23T21:45:05.200Z',
-    updatedAt: '2024-01-23T21:45:05.200Z'
-  },
-  {
-    _id: '65b03361bf20871d3a3aeb9c',
-    text: 'Test note2 content',
-    author: {
-      _id: '6565fd508a848ff2202df79c',
-      firstName: 'User',
-      lastName: 'Test',
-      role: 'tutor'
-    },
-    isPrivate: false,
-    cooperation: '65afbd053d67b51996a67c4c',
-    createdAt: '2024-01-23T21:45:05.200Z',
-    updatedAt: '2024-01-23T21:45:05.200Z'
-  }
-]
+const cooperationId = '675b37915a23a358ab40bf66'
 
-const mockResponseData = {
-  data: ''
+const newNotePayload = {
+  text: 'Newly created note',
+  isPrivate: false
 }
-const userMock = {
-  _id: '648850c4fdc2d1a130c24aea',
-  role: 'tutor',
-  firstName: 'Test',
-  lastName: 'User',
-  photo: 'https://www.google.com'
+
+const updatedNotePayload = {
+  text: 'updated note',
+  isPrivate: false
 }
 
-const mockGetNote = vi.fn()
-const mockUpdateNote = vi.fn().mockReturnValue(mockResponseData)
-const mockCreateNote = vi.fn().mockReturnValue(mockResponseData)
-const mockDeleteNote = vi.fn().mockReturnValue(mockResponseData)
+const duplicatedNote = {
+  ...mockUpdatedNotesData[1],
+  _id: '65b03361bf20871d3adead9c',
+  text: mockUpdatedNotesData[1].text
+}
 
-vi.mock('~/services/cooperation-service', async () => {
-  const actual = await vi.importActual('~/services/cooperation-service')
+vi.mock('react-router-dom', async () => {
+  const actual = await vi.importActual('react-router-dom')
   return {
     ...actual,
-    CooperationNotesService: {
-      getNotes: () => mockGetNote(),
-      updateNote: () => mockUpdateNote(),
-      createNote: () => mockCreateNote(),
-      deleteNote: () => mockDeleteNote()
-    }
+    useParams: () => ({
+      id: cooperationId
+    })
   }
 })
 
+const userMock = {
+  _id: '648850c4fdc2d1a130c24aea',
+  role: 'tutor',
+  firstName: 'Test',
+  lastName: 'User',
+  photo: ''
+}
+
 const appMain = {
   appMain: { userRole: 'tutor', userId: mockNotesData[0].author._id }
 }
 
+const getCooperationUrl = `${URLs.cooperations.getById.replace(':id', cooperationId)}`
```

Comment:
It's already a string, we don't need to wrap a string into string)
```suggestion
const getCooperationUrl = URLs.cooperations.getById.replace(':id', cooperationId)
```